Refactor support for text alignment options in Mail compose.

megangardner opened the pull request:
#### 9d9c2ca3d33d524d768117cc85d83a7287037b3c
<pre>
Refactor support for text alignment options in Mail compose.
<a href="https://bugs.webkit.org/show_bug.cgi?id=291873">https://bugs.webkit.org/show_bug.cgi?id=291873</a>
<a href="https://rdar.apple.com/148599632">rdar://148599632</a>

Reviewed by Wenson Hsieh.

Just moving some bits around.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _alignCenter:]):
(-[WKWebView _alignJustified:]):
(-[WKWebView _alignLeft:]):
(-[WKWebView _alignRight:]):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:

Canonical link: <a href="https://commits.webkit.org/293963@main">https://commits.webkit.org/293963@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e0ec96580ffeceb8fce99dbb955d9abfec2eb04

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100398 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20050 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10349 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105534 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50986 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20356 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28524 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76442 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33498 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103405 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15590 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90690 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56797 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15407 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50355 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85321 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8772 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107889 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27516 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20186 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85396 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27879 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86890 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84934 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21612 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29617 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7349 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21439 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27451 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32692 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27262 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30580 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28820 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
```diff
@@ -226,13 +226,13 @@ typedef std::pair<WebKit::InteractionInformationRequest, InteractionInformationC
+    M(alignCenter) \
+    M(alignJustified) \
+    M(alignLeft) \
+    M(alignRight) \
```
I would remove the trailing \ because it's a hazard to our future selves!

hortont424 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -3367,6 +3367,24 @@ - (void)_action:(id)sender \
+- (IBAction)_alignCenter:(id)sender
+{
+    [self alignCenter:sender];
+}
```
newlines between these

hortont424 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -3367,6 +3367,24 @@ - (void)_action:(id)sender \
 
 #undef FORWARD_ACTION_TO_WKCONTENTVIEW
 
+
```
one less newline here

whsieh reviewed (commented):
Seems reasonable, though I would've also expected a change in `-canPerformActionForWebView:withSender:` to include the new selectors.

(It might also be good to adjust an existing API test to exercise the new methods)

megangardner committed:
Refactor support for text alignment options in Mail compose.
https://bugs.webkit.org/show_bug.cgi?id=291873
rdar://148599632

Reviewed by Wenson Hsieh.

Just moving some bits around.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _alignCenter:]):
(-[WKWebView _alignJustified:]):
(-[WKWebView _alignLeft:]):
(-[WKWebView _alignRight:]):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:

Canonical link: https://commits.webkit.org/293963@main
